src/t_stream.c

```diff
@@ -2880,17 +2903,29 @@ void xpendingCommand(client *c) {
     long long count = 0;
     long long minidle = 0;
     int startex = 0, endex = 0;
+    int mkgroup = 0;   
 
+    robj *o = lookupKeyRead(c->db,key);
+    streamCG *group;
+
+    if (checkType(c,o,OBJ_STREAM)) return;
+    if (o == NULL) {
+        addReplyErrorFormat(c, "No such key '%s'",
+                                     (char*)key->ptr);
+        return;
+    }
+ 
     /* Start and stop, and the consumer, can be omitted. Also the IDLE modifier. */
-    if (c->argc != 3 && (c->argc < 6 || c->argc > 9)) {
+    if (c->argc == 5 || c->argc > 10) {
         addReplyErrorObject(c,shared.syntaxerr);
         return;
     }
 
-    /* Parse start/end/count arguments ASAP if needed, in order to report
-     * syntax errors before any other error. */
-    if (c->argc >= 6) {
-        int startidx = 3; /* Without IDLE */
+    if(c->argc == 4) {
+	mkgroup = 1;
+    } else if (c->argc >= 6) {
+	int totalOption = c->argc-1;
+	int startidx = 3; /* Without IDLE */
```

Comment:
indentation issues..

why do we set `mkgroup` when argc==4?
i don't like this assumption, it can break in the future.
i understand that the current command syntax forces us to count things in order to guess what's coming next, but at least let's actually match strings that we can.